src/server/visual-diff-plugin.js

```diff
@@ -191,36 +191,37 @@ export function visualDiff({ updateGoldens = false, runSubset = false } = {}) {
 					path: updateGoldens ? goldenFileName : screenshotFileName
 				});
 
+				if (updateGoldens) {
+					return { pass: true };
+				}
+
 				const rootLength = join(rootDir, PATHS.VDIFF_ROOT).length + 1;
+
+				const screenshotImage = PNG.sync.read(await readFile(screenshotFileName));
```

Comment:
Set information about the new screenshot early, before we bail if the golden is missing.